Feature: Redid set as background flyout

puppetsw opened the pull request:
<!-- 
🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
I ACKNOWLEDGE THE FOLLOWING BEFORE PROCEEDING:
1. PR may be deleted if it is not following the template
2. Try not to make duplicates. Do a quick search before posting
3. Add a clarified title
-->

**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #7834

**Details of Changes**
Add details of changes here.
- Added set as flyout to toolbar.
- Set as desktop slideshow shows when multiple images as selected.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
![Screenshot 2022-08-20 212506](https://user-images.githubusercontent.com/79826944/185745037-4dedbb77-7ab5-4957-ac45-b300174d6108.png)
![Screenshot 2022-08-20 212449](https://user-images.githubusercontent.com/79826944/185745050-7c991eff-5e5a-4565-9727-9c1b4ec686e6.png)

yaichenbaum commented on builds/azure-pipelines-release.yml:
```diff
@@ -32,7 +32,7 @@ variables:
   value: '**/*.sln'
 - name: buildPlatform
-  value: 'x86|x64|arm64'
+  value: 'x64|arm64'
```
Can you revert this change?

puppetsw replied:
Should be fixed.

yaichenbaum commented on src/Files.Uwp/Strings/en-US/Resources.resw:
```diff
@@ -2792,4 +2789,13 @@ We use App Center to track which settings are being used, find bugs, and fix cra
+    <value>Set as background</value>
+  </data>
+  <data name="SetAsLockscreen" xml:space="preserve">
+    <value>Set as lockscreen background</value>
```
```suggestion
    <value>Lockscreen</value>
```

yaichenbaum commented on src/Files.Uwp/Strings/en-US/Resources.resw:
```diff
@@ -2792,4 +2789,13 @@ We use App Center to track which settings are being used, find bugs, and fix cra
+    <value>Set as lockscreen background</value>
+  </data>
+  <data name="SetAsBackground" xml:space="preserve">
+    <value>Set as desktop background</value>
```
```suggestion
    <value>Desktop</value>
```

puppetsw replied:
Should the Slideshow one be changed too?

yaichenbaum replied:
I think slideshow should stay as is. Just verifying, when multiple images are selected, the only option shown is the slideshow button, is that correct?

puppetsw replied:
No, it currently shows all options, but it will only apply the function to the first selected. I can change that though.

yaichenbaum commented on src/Files.Uwp/UserControls/InnerNavigationToolbar.xaml:
```diff
@@ -272,13 +274,42 @@
+                                </MenuFlyoutItem.Icon>
+                            </MenuFlyoutItem>
+                            <MenuFlyoutItem
+                                x:Name="SetAsSlideshowFlyoutItem"
```
Looking really good! Do you think this should be a dedicated toolbar button so that users don't have to open the flyout to access a single option?

puppetsw replied:
I like that idea If I'm understanding correctly. When you select one image it shows the Flyout for lockscreen and background, but when you select multiple it hides the flyout and replaces with single Slideshow button.

yaichenbaum replied:
That's exactly what I'm imagining 👀

yaichenbaum replied:
I like to call this the intuitive toolbar

puppetsw replied:
I'll make those changes.

puppetsw commented:
![recordingexample](https://user-images.githubusercontent.com/79826944/185817179-212d9040-3909-4a1d-82c1-519879080ce0.gif)

I reverted some changes to the strings and added an extra for Lockscreen, as I think it makes more sense have individual ones for Lockscreen/Desktop instead of replacing SetAs_

yaichenbaum commented on src/Files.Uwp/Strings/en-US/Resources.resw:
```diff
@@ -2786,6 +2783,15 @@ We use App Center to track which settings are being used, find bugs, and fix cra
+  <data name="SetAsBackgroundFlyout" xml:space="preserve">
+    <value>Set as background</value>
+  </data>
+  <data name="SetAsLockscreen" xml:space="preserve">
+    <value>Set as lockscreen</value>
+  </data>
+  <data name="SetAsBackground" xml:space="preserve">
+    <value>Set as desktop background</value>
```
Are these still being used?

yaichenbaum replied:
Ah, are these for the right click menu?

puppetsw replied:
I believe so

yaichenbaum replied:
We should bring the right click menu in line with the toolbar but this is fine for now.

puppetsw replied:
It's next on the list.

yaichenbaum commented:
> ![recordingexample1](https://user-images.githubusercontent.com/79826944/185817713-3fef703f-443e-416a-88dc-e50872622e65.gif) [ ![recordingexample1](https://user-images.githubusercontent.com/79826944/185817713-3fef703f-443e-416a-88dc-e50872622e65.gif) ](https://user-images.githubusercontent.com/79826944/185817713-3fef703f-443e-416a-88dc-e50872622e65.gif) [ ](https://user-images.githubusercontent.com/79826944/185817713-3fef703f-443e-416a-88dc-e50872622e65.gif)
> 
> I reverted some changes to the strings and added an extra for Lockscreen, as I think it makes more sense have individual ones for Lockscreen/Desktop instead of replacing SetAs_

This is perfect!

yaichenbaum reviewed (approved):
Excellent work! LGTM
